rllib/core/learner/learner.py

```diff
@@ -111,12 +112,51 @@ class LearnerHyperparameters:
 
     Validation of the values of these hyperparameters should be done by the
     respective AlgorithmConfig class.
+
+    For configuring different learning behaviors for different (single-agent) RLModules
+    within the Learner, RLlib uses the `per_module_overrides` property (dict), mapping
+    ModuleID to a "sub-version" of a `LearnerHyperparameters` instance, in which the
+    module-specific override settings are applied AND the `per_module_overrides` is set
+    to None.
     """
 
-    # TODO (Sven): Move lr from - currently - optimizer config to only exist here.
-    # lr: float = None
+    # Parameters used for gradient postprocessing (clipping) and gradient application.
+    optimizer_type: str = None
+    learning_rate: LearningRateType = None
+    grad_clip: float = None
+    grad_clip_by: str = None
+
+    # Holds hyperparameters per module. This is not None only in the top-level
+    # Learner's self.hps (whose self.module is a `MARLModule`) and then contains the
+    # correct mappings from ModuleID to the .
+    _per_module_overrides: Optional[Dict[ModuleID, "LearnerHyperparameters"]] = None
+
+    def get_hps_for_module(self, module_id: ModuleID) -> "LearnerHyperparameters":
```

Comment:
New API inside LearnerHPs to get module-specific HP sub-instance.

Reply:
You wouldn't call it with a module_id=None. It's a mandatory arg.

Reply:
Also, some module_ids might not want to override anything, so they'll not be found in the mapping (and just use the top-level HPs object). So no error should be thrown here.

Reply:
Leaving as-is. Reasoning:

self.hps (top-level HPs object):
    get_hps_for_module(abc) -> "I don't have overrides for abc -> return myself"
    get_hps_for_module(abc) -> "I don't have any overrides (for any module ID) -> return myself
    get_hps_for_module(abc) -> "I have an override for abc -> return that"

lower level HPs objects (stored in self._per_module_overrides):
    User would NOT call this method, ever. You would always only call it on (top-level) self.hps instance.